language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/AbstractCompletionProvider.java

```diff
@@ -607,4 +613,60 @@ private CompletionItem getModuleCompletionItem(String label, String insertText,
 
         return completionItems;
     }
+
+    protected List<LSCompletionItem> getNamedArgCompletionItems(BallerinaCompletionContext context,
+                                                            List<ParameterSymbol> params,
+                                                            SeparatedNodeList<FunctionArgumentNode> argumentNodeList) {
+        int cursorPosition = context.getCursorPositionInTree();
+        List<String> existingNamedArgs = new ArrayList<>();
+        int index = 1;
+        for (Node child : argumentNodeList) {
+            TextRange textRange = child.textRange();
+            int startOffset = textRange.startOffset();
+            int endOffset = textRange.endOffset();
+            if (startOffset > cursorPosition
+                    && child.kind() == SyntaxKind.POSITIONAL_ARG || child.kind() == SyntaxKind.REST_ARG) {
+                return Collections.emptyList();
+            } else if ((startOffset > cursorPosition || endOffset < cursorPosition)) {
+                if (child.kind() == SyntaxKind.NAMED_ARG) {
+                    existingNamedArgs.add(((NamedArgumentNode) child).argumentName().name().text());
+                } else if (child.kind() == SyntaxKind.POSITIONAL_ARG && index - 1 < params.size()) {
+                    ParameterSymbol parameterSymbol = params.get(index - 1);
+                    existingNamedArgs.add(parameterSymbol.getName().orElse(""));
+                }
+            }
+            index++;
+        }
+
+        List<LSCompletionItem> completionItems = new ArrayList<>();
+        Predicate<ParameterSymbol> predicate = parameter -> parameter.paramKind() == ParameterKind.REQUIRED
+                || parameter.paramKind() == ParameterKind.DEFAULTABLE;
+        params.stream().filter(predicate).forEach(parameterSymbol -> {
+            Optional<String> paramName = parameterSymbol.getName();
+            TypeSymbol paramType = parameterSymbol.typeDescriptor();
+            String defaultValue = CommonUtil.getDefaultValueForType(paramType).orElse("\"\"");
+            if (paramName.isEmpty() || paramName.get().isEmpty() || existingNamedArgs.contains(paramName.get())) {
+                return;
+            }
+            String label = paramName.get();
+            String insertText = paramName.get() + " = ${1:" + defaultValue + "}";
+            CompletionItem completionItem = NamedArgCompletionItemBuilder.build(label, insertText);
+            completionItems.add(new SymbolCompletionItem(context, parameterSymbol, completionItem));
+        });
+        return completionItems;
+    }
+
+    protected boolean withInNamedArgAssignmentContext(BallerinaCompletionContext context) {
```

Comment:
Shouldn't this be `within` instead of `withIn` since we're checking whether it's **within** the `NamedArgAssignmentContext`?